src/bundle/Controller/SuggestionController.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\Search\Controller;
+
+use Ibexa\Search\Model\SuggestionQuery;
+use Ibexa\Search\Service\SuggestionService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class SuggestionController extends AbstractController
+{
+    private SuggestionService $suggestionService;
+
+    private SerializerInterface $serializer;
+
+    public function __construct(
+        SerializerInterface $serializer,
+        SuggestionService $suggestionService
+    ) {
+        $this->suggestionService = $suggestionService;
+        $this->serializer = $serializer;
+    }
+
+    public function suggestAction(Request $request): JsonResponse
+    {
+        $query = $request->get('query');
+        $limit = (int) $request->get('limit');
+        $language = $request->get('language');
+
+        $suggestionQuery = new SuggestionQuery($query, $limit, $language);
+        $result = $this->suggestionService->suggest($suggestionQuery);
+
+        $serializedResults = $this->serializer->serialize($result, 'json');
```

Comment:
You are not injecting a specialized serializer, so this defaults to App's serializer, which we almost certainly don't want to use (since we need to ensure data normalization shape, and can't impact user's serializer performance).

Consider creating a dedicated serializer instance, or using the REST one (there is one there already).